src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -133,87 +140,103 @@ private static String sanitizeParam(String param) {
 
         LOGGER.info("Run security analysis on contingency lists: {}", context.getContingencyListNames().stream().map(SecurityAnalysisWorkerService::sanitizeParam).collect(Collectors.toList()));
 
-        Mono<Network> network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids());
+        if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
+            return Mono.empty();
+        }
+
+        LOGGER.info("Loading networks");
+
+        Mono<Network> network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids(), resultUuid);
+
+        if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
+            return Mono.empty();
+        }
+
+        LOGGER.info("Loading contingencies");
 
         Mono<List<Contingency>> contingencies = Flux.fromIterable(context.getContingencyListNames())
                 .flatMap(contingencyListName -> actionsService.getContingencyList(contingencyListName, context.getNetworkUuid()))
                 .collectList();
 
         return Mono.zip(network, contingencies)
                 .flatMap(tuple -> {
-                    SecurityAnalysis securityAnalysis = configService.getSecurityAnalysisFactory().create(tuple.getT1(), LocalComputationManager.getDefault(), 0);
-                    CompletableFuture<SecurityAnalysisResult> future = securityAnalysis.run(VariantManagerConstants.INITIAL_VARIANT_ID, context.getParameters(), n -> tuple.getT2());
-                    if (resultUuid != null) {
-                        futures.put(resultUuid, future);
-                    }
                     if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
                         return Mono.empty();
```

Comment:
I think we should add a log to all the short circuit returns because without it, the log will look like this:

```
Run security analysis on contingency lists: ...
loading resource network
loading resource ...
cancelling 
... tons of network-store-client logs because the operation continues...
...and nothing here to remind us that we didn't run the security analysis